sandbox/libs/authn/docs/openapi.yaml

```diff
@@ -0,0 +1,79 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: OpenSearch Identity Permission API
+  license:
+    name: Apache 2.0
+jsonSchemaDialect:
+paths:
+  cluster/permissions:
+    get:
+      requestBody:
+        description: Get a list of all permissions granted to a given principal
+        content:
+          application/json:
+            name: principal name
```

Comment:
@peternied I'd like to get your 2 cents as well. I think we should stick to using `username` as the primary identifier of a user. `username` is a principal (the primary principal in this case) of a user and for readability I think `username` should be used throughout.

Reply:
I swapped it!